lib/jsdom/browser/BrowsingContext.js

```diff
@@ -0,0 +1,58 @@
+"use strict";
+
+const idlUtils = require("../living/generated/utils");
+
+// https://html.spec.whatwg.org/#windows
+class BrowsingContext {
+  // readonly windowProxy: WindowProxy
+  // readonly sessionHistory: SessionHistory
+
+  // create a new browsing context
+  constructor() {
+    this.windowProxy = new WindowProxy();
+    this.sessionHistory = new SessionHistory();
+
+    // 2. Call the JavaScript InitializeHostDefinedRealm() abstract operation
+    const window = new Window();
+    // TODO: InitializeHostDefinedRealm({globalObject: window, globalThisValue: this.windowProxy, relmExecutionContext: new JavaScriptExecutionContext})
+
+
+    this.windowProxy = windowProxy;
+    // https://html.spec.whatwg.org/#session-history
+    this._sessionHistory = [{
+      document: idlUtils.implForWrapper(this._document),
+      url: idlUtils.implForWrapper(this._document)._URL,
+      stateObject: null
+    }];
```

Comment:
So currently all the history-tracking is done by the History object, which hangs off of the window. I guess we need to disentangle them so that session history survives navigation. Then History will be an interface into a SessionHistory.